lib/pkgtailor.js

```diff
@@ -132,6 +120,63 @@ const options = {
 
 // ============================================================================
 
+/**
+ * compose AWS tailoring state string for given Node.js version
+ * @param {number} nodeVer the Node.js major version number
+ */
+function makeAwsTailorStateName(nodeVer) {
+	return `AWS Lambda with Node ${nodeVer}.x`;
+}
+
+/**
+ * construct AWS commandline paramter definition for the given Node.js version number
+ * @param {number} nodeVer Node.js major version number
+ */
+function makeAwsLambdaCmdLineDef(nodeVer) {
+	let deletePattern = makeDeleteFilePattern(nodeVer);
+
+	let environment = {
+		name: makeAwsTailorStateName(nodeVer),
+		deleteFilePredicate: (fn) => {
+			return deletePattern.test(fn) || c32BitPathPattern.test(fn);
+		}
+	};
+
+	return {
+		name: `--AwsLambdaV${nodeVer}`,
+		verbose: `Tailor environment for AWS Lambda with Node.js V${nodeVer}.x`,
+		handler: () => {
+			debug(`setting environment to AWS cLambdaV${nodeVer}EnvDef`);
+			options.environment = environment;
+		}
+	};
+}
+
+/**
+ * returns regexp matching all native extension names but the given API version
+ * @param {number} nodeVer Node.js major version number
+ */
+function makeDeleteFilePattern(nodeVer) {
+	const cNodeToApiVersionMap = {
+		"4": "46",
+		"6": "48",
+		"8": "57"
+	};
+	
+	const apiVer = cNodeToApiVersionMap[nodeVer];
+	if (apiVer.length !== 2) {
+		throw new Error("expected 2 digit api version, got '" + apiVer + "'");
+	}
+	
+	const x = apiVer.charAt(0);
+	const y = apiVer.charAt(1);
+	const patternStr = `_(${x}[^${y}]|[^${x}][0-9])\\.node$`;
```

Comment:
I think this could be simplified to `_(?!${apiVer}).{${apiVer.length}}\\.node$` and as a result we could also change the check above to `apiVer === undefined`